ortho_seq_code/orthogonal_polynomial.py

```diff
@@ -17,9 +17,9 @@ def create_dir_if_not_exists(out_dir):
         ct = 0
         while os.path.exists(out_dir):
             if ct != 0:
-                loc = out_dir.rfind("[") - len(out_dir)
+                loc = -1 * len(str(ct))
```

Comment:
I'm a little confused here. It looks like it's changing the output_dir then having to handle resetting it below this, then adding the new number to it. Why not create an `new_out_dir` variable before the while loop, set is to `output_dir[:]` to start, then on each loop set `new_out_dir = f"out_dir{ct}` (or whatever name format you want). Then you don't have to track this -1 * count string length offset.

Reply:
I think that might work too.. @nkoix added this part in a previous branch.

Reply:
@nkoix is adding this in the plot branch right now, so this is good to go.